src/Persistence_matrix/include/gudhi/persistence_interval.h

```diff
@@ -88,140 +88,164 @@ struct Persistence_interval {
  * @tparam Event_value Second template parameter of @ref Gudhi::persistence_matrix::Persistence_interval.
+ * @param i Interval from which the value should be returned.
+ * @return Either the birth value if @p I == 0, the death value if @p I == 1 or the dimension if @p I == 2.
  */
-template <typename Dimension, typename Event_value>
-struct tuple_size<Gudhi::persistence_matrix::Persistence_interval<Dimension, Event_value> >
-    : integral_constant<size_t, 3> {};
+template <size_t I, typename Dimension, typename Event_value>
+constexpr auto&& get(Gudhi::persistence_matrix::Persistence_interval<Dimension, Event_value>&& i) noexcept {
+  return std::move(i).template get<I>();
+}
 
 /**
  * @ingroup persistence_matrix
  *
- * @brief Partial specialization of `std::tuple_element` for @ref Gudhi::persistence_matrix::Persistence_interval.
- *
- * @tparam I Index of the type to store: 0 for the birth value type, 1 for the death value type and 2 for the
- * dimension value type.
+ * @brief Partial specialization of `get` for @ref Gudhi::persistence_matrix::Persistence_interval.
+ * 
+ * @tparam I Index of the value to return: 0 for the birth value, 1 for the death value and 2 for the dimension.
  * @tparam Dimension First template parameter of @ref Gudhi::persistence_matrix::Persistence_interval.
  * @tparam Event_value Second template parameter of @ref Gudhi::persistence_matrix::Persistence_interval.
+ * @param i Interval from which the value should be returned.
+ * @return Either the birth value if @p I == 0, the death value if @p I == 1 or the dimension if @p I == 2.
  */
-template <size_t I, typename Dimension, typename Event_value>
-struct tuple_element<I, Gudhi::persistence_matrix::Persistence_interval<Dimension, Event_value> > {
-  static_assert(I < 3, "Value mismatch at argument 1 in template parameter list. Maximal possible value is 2.");
-
-  using type = typename conditional <I < 2, Event_value, Dimension>::type;
-};
-
-template <size_t I, typename Dimension, typename Event_value>
-constexpr auto& get(Gudhi::persistence_matrix::Persistence_interval<Dimension, Event_value>& i) noexcept {
-  return Gudhi::persistence_matrix::get<I>(i);
-}
-
-template <size_t I, typename Dimension, typename Event_value>
-constexpr const auto& get(const Gudhi::persistence_matrix::Persistence_interval<Dimension, Event_value>& i) noexcept {
-  return Gudhi::persistence_matrix::get<I>(i);
-}
-
-template <size_t I, typename Dimension, typename Event_value>
-constexpr auto&& get(Gudhi::persistence_matrix::Persistence_interval<Dimension, Event_value>&& i) noexcept {
-  return Gudhi::persistence_matrix::get<I>(move(i));
-}
-
 template <size_t I, typename Dimension, typename Event_value>
 constexpr const auto&& get(const Gudhi::persistence_matrix::Persistence_interval<Dimension, Event_value>&& i) noexcept {
-  return Gudhi::persistence_matrix::get<I>(move(i));
+  return std::move(i.template get<I>());
```

Comment:
I was fine with not changing anything, but now you changed the other one (`&&`), it is inconsistent with here (`const&&`).